src/modules/NodeCollection.js

```diff
@@ -0,0 +1,81 @@
+export class NodeCollection {
+  constructor (nodes) {
+    this.nodes = nodes
+  }
+
+  _isUndefined (value) {
+    return value === undefined || value === null || value === ''
+  }
+
+  _strValue (str, caseSensitive) {
+    return caseSensitive ? `${str}` : `${str}`.toLowerCase()
+  }
+
+  _createStringFilter (filterValue, caseSensitive) {
+    if (this._isUndefined(filterValue)) {
+      filterValue = ''
+    }
+    const strFilter = this._strValue(filterValue, caseSensitive)
+    return value => this._strValue(value, caseSensitive).indexOf(strFilter) >= 0
+  }
+
+  _filterByProps (node, properties, filter) {
+    const mergedProps = [properties].reduce(
+      (merged, prop) => merged.concat(prop),
+      []
+    )
+    return mergedProps.find(prop => filter(node[prop]))
+  }
+
+  filter (properties, filter) {
+    const filtered = this.nodes.filter(node =>
+      this._filterByProps(node, properties, filter)
+    )
+    return new NodeCollection(filtered)
+  }
+
+  contains (properties, value, caseSensitive = false) {
+    return this.filter(
+      properties,
+      this._createStringFilter(value, caseSensitive)
+    )
+  }
+
+  equals (properties, value) {
+    return this.filter(properties, nodeValue => value === nodeValue)
+  }
+
+  betweenNumber (properties, minValue, maxValue) {
+    return this.filter(properties, nodeValue =>
+      (this._isUndefined(minValue) || minValue <= nodeValue) &&
+      (this._isUndefined(maxValue) || maxValue >= nodeValue)
+    )
+  }
+
+  betweenDate (properties, minValue, maxValue) {
+    return this.filter(properties, nodeValue =>
+      (this._isUndefined(minValue) || new Date(minValue).getTime() <= nodeValue) &&
```

Comment:
Are you sure `nodeValue` is always a timestamp?

Reply:
No, I'm not sure, but it was while testing and debugging the code. Do you think it can have different data types?
Then I will wrap it again with new Date().

Reply:
> Then I will wrap it again with new Date()

Reply:
Done.
I also moved `new Date(nodeValue)` into a `const nodeValueTime` to prevent creating the same object twice.